ironfish/src/wallet/walletdb/walletdb.ts

```diff
@@ -553,6 +553,48 @@ export class WalletDB {
     }
   }
 
+  async *loadPendingTransactions(
+    account: Account,
+    headSequence: number,
+    tx?: IDatabaseTransaction,
+  ): AsyncGenerator<TransactionValue> {
+    const encoding = new PrefixEncoding(
```

Comment:
I'm curious why we couldn't do `const encoding = this.pendingTransactionHashes.keyEncoding` here. I see that it's not exactly the same. One is of type `[Buffer, [Buffer, number]]` and the other is `[Buffer, number]`. What's with the extra buffer encoding when we initialize `pendingTransactionHashes` in the constructor?

Reply:
I think we could also consider reusing this somehow though now that we are using this encoding in the constructor, `loadPendingTransactions` and `loadExpiredTransactions`

Reply:
Ok I see, looks like we store the transaction hash as part of the key and then store nothing in the value. Curious if there's a reason we do that? Not that we can change it

Reply:
We could potentially reuse `this.pendingTransactionHashes.keyEncoding` and then just add an empty buffer for the transaction hash range?

Reply:
I'll give that a try and see if we can avoid defining a separate prefix

Reply:
I was able to get this to work by specifying the full range of buffer values with `getPrefixesKeyRange`:

```
const encoding = this.pendingTransactionHashes.keyEncoding

const pendingRange = StorageUtils.getPrefixesKeyRange(
      encoding.serialize([account.prefix, [headSequence + 1, Buffer.alloc(4)]]),
      encoding.serialize([account.prefix, [2 ^ 32, Buffer.alloc(4, 255)]]),
)
```